src/reducers/services.js

```diff
@@ -26,33 +28,33 @@ const routes = (state, log, version) => {
 
   switch (type) {
 
-  case 'update-route-plugin':
-  case 'add-route-plugin':
-  case 'remove-route-plugin':
-    const route = state.find((route) => route.id === params.routeId);
-    return [
-      ...state,
-      {
-        ...route,
-        plugins: plugins(route.plugins, log, version)
-      }
-    ];
-  case 'add-service-route':
-    const { name, ...rest } = parseRoute(content);
-    return [
-      ...state.filter(route => route.name !== params.routeName),
-      {name: params.routeName, ...rest, plugins: [] }
-    ];
-  case 'update-service-route': return state.map(state => {
-    if (state._info.id !== content.id) {
+    case 'update-route-plugin':
+    case 'add-route-plugin':
+    case 'remove-route-plugin':
+      const route = state.find((route) => route.name === params.routeName);
+      return [
+        ...state,
+        {
+          ...route,
+          plugins: plugins(route.plugins, log, version)
+        }
+      ];
+    case 'add-service-route':
+      return [
+        ...state.filter(route => route.name !== params.routeName),
+        {...parseRoute(content), plugins: [] }
```

Comment:
This should be equivalent, since an add-service-route action will always include the route name. Previously the name would have actually been an `id`.